skxray/speckle_visibility/speckle_visibility.py

```diff
@@ -0,0 +1,290 @@
+    ROI's in required images.
+
+    Parameters
+    ----------
+    images_sets : array
+        sets of images as an array
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    max_counts : int
+        maximum speckle counts
+    """
+    max_cts = 0
+    for img_set in images_sets:
+        for img in img_set:
+            max_cts = max(max_cts, meas.maximum(img, label_array))
+    return max_cts
+
+
+def roi_pixel_values(image, labels):
+    """
+    This will provide intensities of the ROI's of the labeled array
+    according to the pixel list
+    eg: intensities of the rings of the labeled array
+
+    Parameters
+    ----------
+    image : array
+        image data dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    roi_pix : dict
+        intensities of the ROI's of the labeled array according
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    #labels, indices = corr.extract_label_indices(label_array)
+    label_num = np.unique(labels)[1:]
+
+    #intensity_distribution = {}
+
+    #for n in label_num:
+    #    value = (np.ravel(image_array)[indices[labels == n].tolist()])
+    #    intensity_distribution[n] = value
+
+    return {n: image[labels == n] for n in range(1, np.max(labels))}
+
+
+def time_bin_edges(number=2, number_of_images=50):
+    """
+    This will provide the geometric series for the integration.
+    Last values of the series has to be less than or equal to number
+    of images
+    ex:
+        1, 2, 4, 8, 16, ...
+        1, 3, 9, 27, ...
+
+    Parameters
+    ----------
+    number : int, optional
+        time steps for the integration
+
+
+    number_of_images : int, 50
+        number of images
+
+    Return
+    ------
+    time_bin : list
+        time binning
+
+    Note
+    ----
+    :math ::
+     a + ar + ar^2 + ar^3 + ar^4 + ...
+
+     a - first term in the series
+     r - is the common ratio
+    """
+
+    time_bin = [1]
+
+    while time_bin[-1]*number < number_of_images:
+        time_bin.append(time_bin[-1]*number)
+    return time_bin
+
+
+def mean_intensity_sets(images_set, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for image sets
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int_labels : dict
```

Comment:
The dict come back for separate image sets.
When they take different image sets( ex: 100 images, 200 images, 150 images, etc..) in different places.